src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AutoCrafterListener.java

```diff
@@ -58,6 +65,21 @@ public void onInteract(PlayerRightClickEvent e) {
                     return;
                 }
 
+                // Check if the recipe of the item is disabled.
+                boolean unlocked = true;
+                for (Recipe recipe : Bukkit.getRecipesFor(e.getItem())) {
```

Comment:
`Bukkit.getRecipesFor()` is really slow and cause issues sometimes, better use our cached version using `SlimefunPlugin.getMinecraftRecipeService().getRecipesFor...` which is faster and safer to use.

Reply:
Nice, I will change it.